scripts/pasha_main.coffee

```diff
@@ -112,6 +120,109 @@ module.exports = (robot) ->
+                        @confirmMsg.send("Failed to create channel #{channelName}: #{err || data.error}")
+                        if data?.error == 'name_taken'
+                            createChannel(baseName, tryNum + 1)
+            # TODO: on other errors, default to #developers
+            createChannel "prio1-#{dateformat(new Date(), 'yyyy-mm-dd')}"
+
+        fiveMinutes: () =>
+            @loadState()
+            return unless @state.prio1?
+            # TODO: configurable link to the status page admin interface
+            @remind 'comm', 'Please update the green/yellow/red status of components on the status page.'
+            @remind 'support', 'Please double-check the updates on the status page, and start public communication if you have a relevant pre-approved message.'
+            @next @tenMinutes, 5
+
+        tenMinutes: () =>
+            @loadState()
+            @remind 'comm', "Please provide an ETA and a simple status update with '#{botName} status ...'\nBad example: packet loss between data centers.\nGood example: network issues, it may be out of our control."
+            # TODO: auto-assign marketing lead
+            @remind 'support', "Ask for clarifications if needed. If the ETA is above 5 minutes, please alert marketing and start textual public communication. Work with @#{@state.prio1.role.comm} to make sure the communicated information is accurate."
+            @tenMinuteIntervalId = setInterval(@everyTenMinutes, 10 * 60 * 1000)
+            @next @sixtyMinutes, 50
+
+        everyTenMinutes: () =>
+            @loadState()
+            @remind 'comm', "Please provide an ETA and a simple status update with '#{botName} status ...'"
+            @remind 'support', "Ask for clarifications if needed. Work with marketing and @#{@state.prio1.role.comm} to provide a public update."
+
+        sixtyMinutes: () =>
+            @loadState()
+            @remind 'marketing', "The prio1 has been going on for an hour. It's time to call a crisis communication meeting, prepare a reactive statement and messaging."
+            @next @twoHours, 60
+
+        twoHours: () =>
+            @loadState()
+            @remind 'marketing', "The prio1 has been going on for two hours. It's time to work with our Brand Communications director to alert HOX."
+
+        stop: () =>
+            @loadState()
+            @remind 'support', 'If possible, please verify that the prio1 is over. If not possible, please acknowledge that you understand engineering believes the prio1 is over.'
+            @remind 'support', "Please work with @#{@state.prio.role.marketing} to update public communications channels, resolve the outstanding issue and update the green/yellow/red status on the status page. Ask @#{@state.prio1.role.comm} for clarifications as needed."
+            clearTimeout @nextTimeoutId if @nextTimeoutId?
+            clearInterval @tenMinuteIntervalId if @tenMinuteIntervalId?
```

Comment:
Nice and neat. Only one thing: it feels a bit weird to have the next timeout defined in all step (eg. I know it's unlikely, but what happens if we redeploy pasha - or it just dies and have to be restarted - during an outage). I'd rather store the next event in brain and check it in - let's say - every minutes. But of course, I can live with this solution (it's way better than not having this :smile:)